testsuite/cluster-test/src/aws.rs

```diff
@@ -112,3 +116,24 @@ fn current_instance_id() -> String {
     let response = response.expect("Metadata request failed");
     response.text().expect("Failed to parse metadata response")
 }
+
+pub fn upload_to_s3(local_filename: &str, bucket: &str, dest_filename: &str) -> Result<()> {
+    let mut f = File::open(local_filename).unwrap();
+    let mut contents: Vec<u8> = Vec::new();
+    match f.read_to_end(&mut contents) {
+        Err(e) => bail!("Error opening file to send to S3: {}", e),
+        Ok(_) => {
+            let req = PutObjectRequest {
+                bucket: bucket.to_owned(),
+                key: dest_filename.to_owned(),
+                body: Some(contents.into()),
+                ..Default::default()
+            };
+            S3Client::new(Region::UsWest2)
```

Comment:
One q - can we make this a method in `Aws` and have S3Client as member of `Aws` struct?

Reply:
Right now Aws::new is tied to discovery, so we cannot create a new instance here. Other option is to add it to Context, but this seemed simpler to have a standalone function.